lib/Twig/Extension/Core.php

```diff
@@ -1507,12 +1507,15 @@ function twig_get_attribute(Twig_Environment $env, Twig_Source $source, $object,
                 continue;
             }
 
-            if (!isset($cache[$name])) {
-                $cache[$name] = $method;
-                $cache[$lcName] = $method;
+            if (!isset($classCache[$name])) {
+                $classCache[$name] = $method;
+            }
+
+            if (!isset($classCache[$lcName])) {
```

Comment:
@fabpot Since #2316 checks for existing cache entries have been separated as I did it here. Was it intended to merge them for Twig 2.0?